### PR TITLE
Update ESLint + ESLint Plugins

### PR DESCRIPTION
**Pull request by ajcumine**
### Changes
Updates ESLint, related plugins, and linked dependencies.

In doing this a number of new default rules were added to the linting rules, many of which I fixed, however some added to the `.eslintrc` to be fixed later as they would increase the size of this PR significantly while not creating any performance/functional improvements if they are ignored.

These rules are:
```
"jsx-a11y/anchor-is-valid"
"jsx-a11y/click-events-have-key-events"
"jsx-a11y/no-noninteractive-tabindex"
"jsx-a11y/label-has-for"
```

There are also a few instances of eslint being disabled for a line, this can also be addressed in the same future PR.

#### Trello Ticket / Github Issue Link
https://trello.com/c/AkbguCJk/228-update-babel-eslint

### Test
General regression test to ensure functionality remains the same as a large proportion of the code base is affected by this change.

#### Can this PR be merged immediately after testing (eg. rebase, infrastructure update)?
Can be merged after #772 is merged.

### CODE REVIEW

File: site/components/wall/index.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ export default function Wall({ children, cols }: WallProps) {
       {elements.map((column, index) => {
         const size = Math.floor(12 / cols);
         return (
+          // eslint-disable-next-line
```
Comment (dpiatek) [inline site/components/wall/index.js]: As above re:disabling

File: site/pages/join-us/jobs/index.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ type JobsProps = {
 
 export default function Jobs({ jobs }: JobsProps) {
   const listings = jobs.map((job, index) => (
-    <Note key={index}>
+    <Note key={`${job.title + '-' + index}`}>
```
Comment (dpiatek) [inline site/pages/join-us/jobs/index.js]: `${job.title}-${index}` ?

File: dev/generate-sitemap/site-routes.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
 export default function getSiteRoutes(state, routeDefinitions) {
   const allRoutes = [];
+
+  // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (ajcumine) [reply dev/generate-sitemap/site-routes.js]: yes, sorry forgot this 👍

File: site/components/container/index.js [hunk 4]
```diff
@@ -8,19 +8,13 @@ import * as React from 'react';
...
 type ContainerProps = {
-  children?: React.Node,
+  children: React.Node,
 };
 
-export default class Container extends React.Component<ContainerProps> {
-  static defaultProps = {
-    children: [],
-  };
+const Container = ({ children }: ContainerProps) => (
```
[the remark this answers is not in the document]
Comment (ajcumine) [reply site/components/container/index.js]: all `<Container />` components currently have children and they shouldn't be used if they don't so I removed it as the updated eslint and flow configuration did not like it

**Pull request merged**